sources/logdog/src/main.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+/*!
+# Introduction
+
+`logdog` is a program that gathers logs from various places on a Bottlerocket host and combines them
+into a tarball for easy export.
+
+Usage example:
+```
+$ logdog
+logs are at: /tmp/bottlerocket-logs.tar.gz
+```
+
+# Logs
+
+For the commands used to gather logs, please see [commands.rs](src/commands.rs).
+
+*/
+
+#![deny(rust_2018_idioms)]
+
+mod commands;
+mod create_tarball;
+mod error;
+
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::{env, process};
+
+use commands::commands;
+use create_tarball::create_tarball;
+use error::Result;
+use tempfile::TempDir;
+
+use snafu::{ErrorCompat, ResultExt};
+
+const ERROR_FILENAME: &str = "logdog.errors";
+const OUTPUT_FILENAME: &str = "bottlerocket-logs.tar.gz";
+const TARBALL_DIRNAME: &str = "bottlerocket-logs";
+
+/// Prints a usage message in the event a bad arg is passed.
+fn usage() -> ! {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {}
+            [ --output PATH ]       where to write archived logs
+",
+        program_name,
+    );
+    process::exit(2);
+}
+
+/// Prints a more specific message before exiting through usage().
+fn usage_msg(msg: &str) -> ! {
+    eprintln!("{}\n", msg);
+    usage();
+}
+
+/// Parses the command line arguments.
+fn parse_args(args: env::Args) -> PathBuf {
+    let mut output_arg = None;
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--output" => {
+                output_arg = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --output")),
+                )
+            }
+            _ => usage(),
+        }
+    }
+
+    match output_arg {
+        Some(path) => PathBuf::from(path),
+        None => env::temp_dir().as_path().join(OUTPUT_FILENAME),
+    }
+}
+
+/// Runs a command and writes its output to a file.
+pub(crate) fn run_command<P: AsRef<Path>>(output_filepath: P, command: &str) -> Result<()> {
+    let command_parts: Vec<String> = command
+        .to_owned()
+        .split(" ")
+        .map(|s| s.to_string())
+        .collect::<Vec<String>>();
+    let command = match command_parts.get(0) {
+        Some(c) => c.into(),
+        None => "".to_string(),
+    };
+    let args: Vec<String> = if command_parts.len() > 1 {
+        command_parts[1..].to_owned()
+    } else {
+        vec![]
+    };
```

Comment:
Can we use https://crates.io/crates/shell-words?  No deps!  Shell splitting is hard and it's easy to cause problems, including in security.